packages/@aws-cdk/aws-iot-actions/lib/iot-republish-action.ts

```diff
@@ -0,0 +1,72 @@
+import * as iam from '@aws-cdk/aws-iam';
+import * as iot from '@aws-cdk/aws-iot';
+import { CommonActionProps } from './common-action-props';
+import { singletonActionRole } from './private/role';
+
+/**
+ * MQTT Quality of Service
+ *
+ * @see https://docs.aws.amazon.com/iot/latest/developerguide/mqtt.html#mqtt-qos
+ */
+export enum MqttQos {
+  /**
+   * QoS level 0. Sent zero or more times.
+   * This level should be used for messages that are sent over reliable communication links or that can be missed without a problem.
+   */
+  ZERO_OR_MORE_TIMES,
+
+  /**
+   * QoS level 1. Sent at least one time, and then repeatedly until a PUBACK response is received.
+   * The message is not considered complete until the sender receives a PUBACK response to indicate successful delivery.
+   */
+  AT_LEAST_ONCE,
+}
+
+/**
+ * Configuration properties of an action to republish MQTT messages.
+ */
+export interface IotRepublishActionProps extends CommonActionProps {
+  /**
+   * The Quality of Service (QoS) level to use when republishing messages.
+   *
+   * @see https://docs.aws.amazon.com/iot/latest/developerguide/mqtt.html#mqtt-qos
+   *
+   * @default MqttQos.ZERO_OR_MORE_TIMES
+   */
+  readonly qos?: MqttQos;
+}
+
+/**
+ * The action to put the record from an MQTT message to republish another MQTT topic.
+ */
+export class IotRepublishAction implements iot.IAction {
+  private readonly qos?: MqttQos;
+  private readonly role?: iam.IRole;
+
+  /**
+   * @param topic The MQTT topic to which to republish the message.
+   * @param props Optional properties to not use default
+   */
+  constructor(private readonly topic: string, props: IotRepublishActionProps = {}) {
+    this.qos = props.qos;
+    this.role = props.role;
+  }
+
+  bind(rule: iot.ITopicRule): iot.ActionConfig {
+    const role = this.role ?? singletonActionRole(rule);
+    role.addToPrincipalPolicy(new iam.PolicyStatement({
+      actions: ['iot:Publish'],
+      resources: ['*'],
```

Comment:
Hmm... does this have to be `"*"`? Doesn't the fact that we have the `topic` here allow us to write a more constrained permission policy?

Reply:
I think too, it is maybe few restrictions. But I have no good idea..

When the passed topic is literal, [The topic's ARN `arn:aws:iot:aws-region:AWS-account-ID:topic/Topic`](https://docs.aws.amazon.com/iot/latest/developerguide/topics.html#topicnames-arn) can be identified.
But when the passed topic includes any expressions (e.g. `"${topic()}/republished"`, this case is rather common), the ARN cannot be identified.

Reply:
OK. Let's leave it as-is for now (if you want to add a quick blurb in the ReadMe about this, feel free, but I won't require it).